roles/confluent.common/tasks/secrets_protection.yml

```diff
@@ -51,25 +61,32 @@
     secrets_protection_masterkey: "{{ slurped_masterkey.content|b64decode}}"
   when: not secrets_protection_masterkey
 
-- name: Encrypt All Properties that Contain 'password'
-  shell: |
-    {{ confluent_cli_path }} secret file encrypt --config-file {{ config_path }} \
-      --local-secrets-file {{ secrets_file }} \
-      --remote-secrets-file {{ secrets_file }}
-  environment:
-    CONFLUENT_SECURITY_MASTER_KEY: "{{ secrets_protection_masterkey }}"
-  when: encrypt_passwords|bool
+- name: Create Encyrpt Properties List
+  set_fact:
+    encryrpt_properties: "{{ final_properties | dict2items | map(attribute='key') | select('match', properties_patterns|join('|'))
+      + properties if encrypt_passwords|bool else properties }}"
+  vars:
+    properties_patterns:
+      - '.*password.*'
+      - '.*basic.auth.user.info.*'
+      - 'ldap.java.naming.security.credentials'
+      - 'confluent.license'
+      - '.*sasl.jaas.config'
+
+- name: Debug Properties to Mask
+  debug:
+    var: encryrpt_properties
+  run_once: true
 
-- name: Encrypt Selected Properties
+- name: Encrypt Properties
   shell: |
     {{ confluent_cli_path }} secret file encrypt --config-file {{ config_path }} \
       --local-secrets-file {{ secrets_file }} \
       --remote-secrets-file {{ secrets_file }} \
-      --config {{ item }}
-  loop: "{{ properties }}"
+      --config "{{ encryrpt_properties | join (',') }}"
   environment:
     CONFLUENT_SECURITY_MASTER_KEY: "{{ secrets_protection_masterkey }}"
-  when: properties|length>0
+  changed_when: true
```

Comment:
This looks a bit bogus - at least without a `when`. Are you able to detect if the properties are already encrypted and then skip the task? At least I prefer idempotent playbooks - where everything is green (no changes) on second run. Or am I missing something?

Reply:
So unfortunately this secrets protection logic is not very idempotent bc I need to reset the server.properties file with unmasked props before running this script which then rewrites the file.

But i could add back the when condition bc i suppose someone could say
```
kafka_broker_secrets_protection_encrypt_passwords: false
kafka_broker_secrets_protection_encrypt_properties: []
```
and in that case the `properties|length==0`

Also if you can notice in the next copy task I had to do some fancy logic around triggering the handler. Essentially i compare the old masked file with the new one and if theyre different then trigger the restart.